lib/src/model/library.dart

```diff
@@ -197,9 +197,40 @@ class Library extends ModelElement
     return prefixToLibrary;
   }
 
-  late final String dirName = (isAnonymous ? nameFromPath : name)
-      .replaceAll(':', '-')
-      .replaceAll('/', '_');
+  /// An identifier for this library based on its location.
+  ///
+  /// This provides filename collision-proofing for anonymous libraries by
+  /// incorporating more from the location of the anonymous library into the
+  /// name calculation. Simple cases (such as an anonymous library in 'lib/')
+  /// are the same, but this will include slashes and possibly colons
+  /// for anonymous libraries in subdirectories or other packages.
+  late final String dirName = () {
+    String nameFromPath;
+    if (isAnonymous) {
```

Comment:
Quick question: What are anonymous libraries? When do these occur?

Reply:
An anonymous library is either a library without a `library` directive, or one where the `library` directive has no name (`library;`).